opm/autodiff/ParallelOverlappingILU0.hpp

```diff
@@ -475,11 +939,66 @@ class ParallelOverlappingILU0
         detail::convertToCRS( *ILU, lower_, upper_, inv_ );
     }
 
+    /// \brief Reorder D if needed and return a reference to it.
+    Range& reorderD(const Range& d)
+    {
+        if ( ordering_.empty())
+        {
+            return const_cast<Range&>(d);
```

Comment:
Will any caller ever modify the result of `reorderD()` if `ordering_.empty()`?

Reply:
Yes, in parallel we have to be able to communicate some entries between processors and that requires writing to the vector returned by this function. Here we violate some assumptions in the interface of dune-istl. The const_cast hack has been there before just directly in ::apply.

Reply:
> Here we violate some assumptions in the interface of dune-istl.

And, strictly speaking, of C++.  Mutating a const object (the second parameter of apply()) is undefined behaviour.  If there's no better way of achieving the desired result then we should install into master as-is.  We could possibly add a comment to that effect at the `reorderD()` call site (in apply()).